src/main/java/org/broadinstitute/consent/http/resources/DataSetResource.java

```diff
@@ -110,6 +111,45 @@ public Response createDataset(@Auth AuthUser user, @Context UriInfo info, String
         return Response.created(uri).entity(createdDataset).build();
     }
 
+    @PUT
+    @Consumes("application/json")
+    @Produces("application/json")
+    @Path("/{datasetId}")
+    @PermitAll
+    public Response updateDataset(@Auth AuthUser user, @Context UriInfo info, String json, @PathParam("datasetId") Integer datasetId) {
+        DataSetDTO inputDataset = new Gson().fromJson(json, DataSetDTO.class);
+        if (Objects.isNull(inputDataset)) {
+            throw new BadRequestException("Dataset is required");
+        }
+        if (Objects.isNull(inputDataset.getProperties()) || inputDataset.getProperties().isEmpty()) {
+            throw new BadRequestException("Dataset must contain required properties");
```

Comment:
TOL: we should make sure the swagger doc has an appropriate description about which properties are required.